brave/brave-core
# Launched additional confirmation dialog to the end of support

When user closes infobar with checked state,
additional dialog is launched to confirm user's choice again.
Obsolete system infobar will not be shown again when user clicks OK.

BraveConfirmDialog is introduced to support checkbox to upstream confirm
dialog.

fix https://github.com/brave/brave-browser/issues/27649

![image](https://user-images.githubusercontent.com/6786187/210998624-2bcec186-c6ed-4b1b-b468-4ed0c663b7d9.png)


Resolves 

## Submitter Checklist:

  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run lint`, `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`

## Reviewer Checklist:


## After-merge Checklist:

  changes has landed on

## Test Plan:

1. Launch Brave on Win7/8 and check obsolete system infobar with don't show again check button is shown
2. Don't check `Don't show again` checkbox and click close(`x`) button
3. Check that infobar is hidden and infobar is shown again after relaunch
4. Check don't show again checkbox and click close(`x`) button
5. Check that confirm dialog is shown and click `Cancel` button
6. Check that infobar is hidden and infobar is shown again after relaunch
7. Check don't show again checkbox and click close(`x`) button
8. Check that confirm dialog is shown and click `Ok` button
9. Check that infobar is hidden and infobar is not shown anymore after relaunch
